src/Paramore.Brighter/CommandProcessor.cs

```diff
@@ -564,34 +602,59 @@ public string[] DepositPost<TRequest>(IEnumerable<TRequest> requests, RequestCon
         {
             s_logger.LogInformation("Save bulk requests request: {RequestType}", typeof(TRequest));
             
-            var span = CreateSpan(string.Format(CREATEEVENT, typeof(TRequest).Name));
+            var span = _tracer?.CreateBatchSpan<TRequest>(requestContext?.Span, options: _instrumentationOptions);
             var context = InitRequestContext(span, requestContext);
-
+            
             try
             {
                 var successfullySentMessage = new List<string>();
 
-                foreach (var batch in SplitRequestBatchIntoTypes(requests))
+                foreach (var request in requests)
```

Comment:
This is a significant change in Functionality, Previously we split into types (so that we could pass through the correct request type and the provider factory could correctly identity the request type) and then put them as a bulk operation onto the bus as well as marking them as disposed as a single Bulk Operation.  This will sent them to the bus one by one and then dispatch them one by one

Reply:
So the binding to the type happens now in the call to DepositPost. This means that the DepositPost has the correct request type and can be identified by the mapper factory correctly. LINQ was then used to dispatch. Were we batching up writes to the Outbox? We tend to pay the cost for the pipeline transform call per message.  Batch Clear is a separate issue, as it does not involve the message transform pipeline, so we don't have the same issues.